Android vibration fixes.

MAJigsaw77 opened the pull request:
* Fixed some deprecation warnings.
* Cancel the vibration when the app is being paused or destroyed.

MAJigsaw77 committed:
Update GameActivity.java

MAJigsaw77 committed:
Maintain style

MAJigsaw77 committed:
Cancel the vibration when the app pauses or when is being destroyed.

MAJigsaw77 committed:
Oops

MAJigsaw77 committed:
Oops

MAJigsaw77 commented:
@player-03 Can i get a review on this pr?

player-03 commented on templates/android/template/app/src/main/java/org/haxe/lime/GameActivity.java:
```diff
@@ -137,6 +139,14 @@ protected void onCreate (Bundle state) {
+			vibrator.cancel ();
+
+		}
+
```
Is this necessary? [The documentation](https://developer.android.com/reference/android/os/Vibrator) claims ongoing vibrations will be stopped automatically.

MAJigsaw77 replied:
> Is this necessary? [The documentation](https://developer.android.com/reference/android/os/Vibrator) claims ongoing vibrations will be stopped automatically.

I'll remove it then

MAJigsaw77 committed:
Alright

player-03 committed:
Reduce duplicate logic.

I'd rather duplicate the SDK version check, not the math.

player-03 committed:
Simplify and document vibration pattern logic.

player-03 commented:
Interesting. According to the documentation, [the first entry in the array is the delay before vibration starts](https://developer.android.com/reference/android/os/VibrationEffect#createWaveform\(long[],%20int\)), so they recommend leaving that 0.

The question is, is this delay a bug or a feature? If it's a bug, we could fix it now. If it's a feature, we probably want to add an argument to toggle it on or off.

player-03 committed:
Remove log message.

`GameActivity` doesn't print any other messages, so this seems out of place. Plus, it would happen every time even if there's no vibration ongoing.

MAJigsaw77 commented:
> Interesting. According to the documentation, [the first entry in the array is the delay before vibration starts](https://developer.android.com/reference/android/os/VibrationEffect#createWaveform(long%5B%5D,%20int)), so they recommend leaving that 0.
> 
> The question is, is this delay a bug or a feature? If it's a bug, we could fix it now. If it's a feature, we probably want to add an argument to toggle it on or off.

I don't really know what to say tbh

player-03 committed:
Remove delay before vibration begins.
